### PR TITLE
Moath/at 96/entry pages

### PR DESCRIPTION
**Pull request by Right2Drive**

### CODE REVIEW

**Comment by Right2Drive** [comment]
Just opening a PR so I can leave some comment as discussed yesterday

File: src/view/components/AddEntry/index.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  private onCancelPress = () => this.props.goBack();
+  
+  /** Handle pressing save button */
+  private onSavePress = () => {
+    const timeNow = moment().utc().toString();
+    let newEntry: MilkEntry = {
+      uuid: 'fakeEntryUuid',
+      toUUID: 'fakeToUuid',
+      fromUUID: 'fakeFromUuid',
+      datetime: timeNow,
```
Comment (Right2Drive) [inline src/view/components/AddEntry/index.tsx]: The default `toString` method of moment is probably not going to be the same format Enoch wants it on the backend. We have to match for format `YYYY-MM-DD hh-mm-ss` for anything going to the backend (which will hopefully be validated on the core at some point). I don't know if my capitalization is correct on the various fields btw so don't just stick that in assuming I know what I'm doing :P

File: src/view/components/AddEntry/index.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    const isPrimary = color === 'PRIMARY';
+
+    return (
+      <Col style={Styles.button}>
+        <Button block info={isInfo} primary={isPrimary} onPress={onPress}>
+          <Text>{text}</Text>
+        </Button>
+      </Col>
+    );
+  }
```
Comment (Right2Drive) [inline src/view/components/AddEntry/index.tsx]: Nice! I like how you did this! Looks very clean!

File: src/view/components/AddEntry/index.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+            {this.renderCancelButton()}
+            {this.renderSaveButton()}
+          </Row>
+        </Grid>
+      </Content>
+    );
+  }
+}
+
+const AddEntryPage = new Composer<NestedPropsType>(AddEntry).page;
```
Comment (Right2Drive) [inline src/view/components/AddEntry/index.tsx]: If we aren't creating a fab page we shouldn't inject FabProps

File: src/view/components/DataTable/index.tsx [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
 import { List, ListItem, Text, Grid, Col, Card } from 'native-base';
 import styles from './style';
+import { Route } from '../../navigation/navigator';
 
 
 interface OwnPropsType {
   headers: string[];
   values: any[];
+  route?: Route;
+  onPress?(route: Route): void;
```
Comment (Right2Drive) [inline src/view/components/DataTable/index.tsx]: If one of these being required makes it necessary the other is required too, you can do one of two things:
1) Create an interface that holds a required route and onPress that is itself an optional property on onPress
2) Create an interface (`RoutedPropsType` for this example) that extends the interface with headers/values (we will call this `BasePropsType` for my example) and then `type OwnPropsType = RoutedPropsType | BasePropsType`

File: src/view/components/EditEntry/index.tsx [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+  
+  /** Handle pressing save button */
+  private onSavePress = () => {
+    let newEntry: PartialDairy = {
+      uuid: this.props.milkEntry.uuid,
+      volume: this.state.volume,
+      quality: this.state.quality,
+      costPerUnit: this.state.costPerUnit,
+    };
+    this.props.updateDairy(newEntry);
```
Comment (Right2Drive) [inline src/view/components/EditEntry/index.tsx]: I think we are going with the name `milk` for everything instead of `dairy`

**Comment by Right2Drive** [comment]
@mfalthaw I left some review comments, will close the PR now and you can create a new one or re-open when you see fit

File: src/view/components/AddEntry/index.tsx (hunk 1 above)
Comment (mfalthaw) [reply src/view/components/AddEntry/index.tsx]: @jeinglis @enochtsang We'll have to be in sync with this one.

File: src/view/components/AddEntry/index.tsx (hunk 3 above)
Comment (mfalthaw) [reply src/view/components/AddEntry/index.tsx]: You're right, we're not doing fabs on these pages anymore. Fixed.

File: src/view/components/EditEntry/index.tsx (hunk 5 above)
Comment (mfalthaw) [reply src/view/components/EditEntry/index.tsx]: I think this should auto-refactor when we refactor module `dairy` name, no?

File: src/view/components/DataTable/index.tsx (hunk 4 above)
Comment (mfalthaw) [reply src/view/components/DataTable/index.tsx]: ooh that's clever! done.

**Pull request closed without merging**